### PR TITLE
Pioneer 15 recovered moorings

### PR DESCRIPTION
**Pull request by cdobs**
Ingestion csvs for moorings recovered on Pioneer 15:
CP01CNSM-R00012
CP02PMUO-R00015
CP03ISSM-R00011
CP04OSSM-R00011

### CODE REVIEW

File: CP01CNSM/CP01CNSM_R00012_ingest.csv [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/hyd2/*.hyd2.log,CP01CNSM-SBD12-03-HYDGN0000,recovered_host,Expected,
+mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/pco2a/*pco2a.log,CP01CNSM-SBD12-04-PCO2AA000,recovered_host,Expected,
+mi.dataset.driver.wavss_a.dcl.wavss_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/wavss/*.wavss.log,CP01CNSM-SBD12-05-WAVSSA000,recovered_host,Expected,
+mi.dataset.driver.metbk_a.dcl.metbk_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/metbk2/*metbk2.log,CP01CNSM-SBD12-06-METBKA000,recovered_host,Expected,
+mi.dataset.driver.fdchp_a.dcl.fdchp_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/fdchp/*fdchp.log,CP01CNSM-SBD12-08-FDCHPA000,recovered_host,Expected,
+#mi.dataset.driver.fdchp_a.fdchp_a_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/instruments/dcl12/FDCHP_sn*/D201*/fdchp*.dat,CP01CNSM-SBD12-08-FDCHPA000,recovered_inst,Not Expected,Data unable to be downloaded from instrument
```
Comment (snwhite) [inline CP01CNSM/CP01CNSM_R00012_ingest.csv:47]: I believe the folders in the FDCHP_sn folder that start D201 use the date (year and month)... e.g., D201809.
So this should be changed to D202 or just D20 or it won't read the folders for 2020 and beyond.

File: CP03ISSM/CP03ISSM_R00011_ingest.csv [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+mi.dataset.driver.optaa_dj.dcl.optaa_dj_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl37/optaa2/*.optaa2.log,CP03ISSM-MFD37-01-OPTAAD000,recovered_host,Expected,
+mi.dataset.driver.ctdbp_cdef.dcl.ctdbp_cdef_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl37/ctdbp2/*ctdbp2.log,CP03ISSM-MFD37-03-CTDBPD000,recovered_host,Expected,
+mi.dataset.driver.ctdbp_cdef.ctdbp_cdef_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/instruments/dcl37/CTDBP*/SBE*.hex,CP03ISSM-MFD37-03-CTDBPD000,recovered_inst,Expected,
+mi.dataset.driver.dosta_abcdjm.dcl.dosta_abcdjm_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl37/dosta2/*dosta2.log,CP03ISSM-MFD37-04-DOSTAD000,recovered_host,Expected,
+mi.dataset.driver.zplsc_c.dcl.zplsc_c_dcl_telemetered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl37/zplsc/*.zplsc.log,CP03ISSM-MFD37-07-ZPLSCC000,recovered_host,Expected,
+mi.dataset.driver.zplsc_c.zplsc_c_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/instruments/dcl37/ZPLSC*/*/*/*.01A,CP03ISSM-MFD37-07-ZPLSCC000,recovered_inst,Expected,
```
Comment (snwhite) [inline CP03ISSM/CP03ISSM_R00011_ingest.csv:20]: See comment for CP01CNSM on specifying the DATA folder in the folder mask.

File: CP04OSSM/CP04OSSM_R00011_ingest.csv [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+mi.dataset.driver.optaa_dj.dcl.optaa_dj_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl37/optaa2/*.optaa2.log,CP04OSSM-MFD37-01-OPTAAD000,recovered_host,Expected,
+mi.dataset.driver.ctdbp_cdef.dcl.ctdbp_cdef_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl37/ctdbp2/*ctdbp2.log,CP04OSSM-MFD37-03-CTDBPE000,recovered_host,Expected,
+mi.dataset.driver.ctdbp_cdef.ctdbp_cdef_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/instruments/dcl37/CTDBP*/SBE*.hex,CP04OSSM-MFD37-03-CTDBPE000,recovered_inst,Expected,
+mi.dataset.driver.dosta_abcdjm.dcl.dosta_abcdjm_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl37/dosta2/*dosta2.log,CP04OSSM-MFD37-04-DOSTAD000,recovered_host,Expected,
+mi.dataset.driver.zplsc_c.dcl.zplsc_c_dcl_telemetered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl37/zplsc/*.zplsc.log,CP04OSSM-MFD37-07-ZPLSCC000,recovered_host,Expected,
+mi.dataset.driver.zplsc_c.zplsc_c_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/instruments/dcl37/ZPLSC*/*/*/*.01A,CP04OSSM-MFD37-07-ZPLSCC000,recovered_inst,Expected,
```
Comment (snwhite) [inline CP04OSSM/CP04OSSM_R00011_ingest.csv:20]: See comment for CP01CNSM on specifying the DATA folder in the folder mask.

**Review by snwhite — CHANGES_REQUESTED** [review]
There is one issue for this deployment and a few changes that will relate to all ingestions or re-ingestions of ZPLSC/G data going forward, and FDCHP.

File: CP01CNSM/CP01CNSM_R00012_ingest.csv [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/hyd2/*.hyd2.log,CP01CNSM-SBD12-03-HYDGN0000,recovered_host,Expected,
+mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/pco2a/*pco2a.log,CP01CNSM-SBD12-04-PCO2AA000,recovered_host,Expected,
+mi.dataset.driver.wavss_a.dcl.wavss_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/wavss/*.wavss.log,CP01CNSM-SBD12-05-WAVSSA000,recovered_host,Expected,
+mi.dataset.driver.metbk_a.dcl.metbk_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/metbk2/*metbk2.log,CP01CNSM-SBD12-06-METBKA000,recovered_host,Expected,
+mi.dataset.driver.fdchp_a.dcl.fdchp_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/cg_data/dcl12/fdchp/*fdchp.log,CP01CNSM-SBD12-08-FDCHPA000,recovered_host,Expected,
+#mi.dataset.driver.fdchp_a.fdchp_a_recovered_driver,/omc_data/whoi/OMC/CP01CNSM/R00012/instruments/dcl12/FDCHP_sn*/D20*/fdchp*.dat,CP01CNSM-SBD12-08-FDCHPA000,recovered_inst,Not Expected,Data unable to be downloaded from instrument
```
Comment (snwhite) [inline CP01CNSM/CP01CNSM_R00012_ingest.csv:47]: Updated file looks good.

File: CP03ISSM/CP03ISSM_R00011_ingest.csv [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+mi.dataset.driver.cg_stc_eng.stc.mopak_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl11/mopak/*.mopak.log,CP03ISSM-SBD11-01-MOPAK0000,recovered_host,Expected,
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl11/hyd1/*.hyd1.log,CP03ISSM-SBD11-02-HYDGN0000,recovered_host,Expected,
+mi.dataset.driver.metbk_a.dcl.metbk_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl11/metbk/*.metbk.log,CP03ISSM-SBD11-06-METBKA000,recovered_host,Expected,
+mi.dataset.driver.cg_dcl_eng.dcl.cg_dcl_eng_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl12/syslog/*.syslog.log,CP03ISSM-SBD12-00-DCLENG000,recovered_host,Expected,
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl12/hyd2/*.hyd2.log,CP03ISSM-SBD12-03-HYDGN0000,recovered_host,Expected,
+mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_recovered_driver,/omc_data/whoi/OMC/CP03ISSM/R00011/cg_data/dcl12/pco2a/*pco2a.log,CP03ISSM-SBD12-04-PCO2AA000,recovered_host,Expected,
```
Comment (snwhite) [inline CP03ISSM/CP03ISSM_R00011_ingest.csv:43]: Updated file looks good.

File: CP04OSSM/CP04OSSM_R00011_ingest.csv [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+mi.dataset.driver.cg_stc_eng.stc.mopak_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl11/mopak/*.mopak.log,CP04OSSM-SBD11-01-MOPAK0000,recovered_host,Expected,
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl11/hyd1/*.hyd1.log,CP04OSSM-SBD11-02-HYDGN0000,recovered_host,Expected,
+mi.dataset.driver.metbk_a.dcl.metbk_a_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl11/metbk/*.metbk.log,CP04OSSM-SBD11-06-METBKA000,recovered_host,Expected,
+mi.dataset.driver.cg_dcl_eng.dcl.cg_dcl_eng_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl12/syslog/*.syslog.log,CP04OSSM-SBD12-00-DCLENG000,recovered_host,Expected,
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl12/hyd2/*.hyd2.log,CP04OSSM-SBD12-03-HYDGN0000,recovered_host,Expected,
+mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_recovered_driver,/omc_data/whoi/OMC/CP04OSSM/R00011/cg_data/dcl12/pco2a/*pco2a.log,CP04OSSM-SBD12-04-PCO2AA000,recovered_host,Expected,
```
Comment (snwhite) [inline CP04OSSM/CP04OSSM_R00011_ingest.csv:43]: Updated files looks good.

**Review by snwhite — APPROVED** [review]
Looks good following corrections.

File: CP01CNSM/CP01CNSM_R00012_ingest.csv (hunk 4 above)
Comment (johnmlund) [inline CP01CNSM/CP01CNSM_R00012_ingest.csv:47]: looks good

File: CP02PMUO/CP02PMUO_R00015_ingest.csv [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+mi.dataset.driver.WFP_ENG.STC_IMODEM.wfp_eng_stc_imodem_recovered_driver,/omc_data/whoi/OMC/CP02PMUO/R00015/instruments/CWFP*/*/E*.DA?,CP02PMUO-WFP01-00-WFPENG000,recovered_wfp,Available,
+mi.dataset.driver.vel3d_k.wfp.vel3d_k_wfp_recovered_driver,/omc_data/whoi/OMC/CP02PMUO/R00015/instruments/CWFP*/*/A*.DA?,CP02PMUO-WFP01-01-VEL3DK000,recovered_wfp,Available,
+mi.dataset.driver.dofst_k.wfp.coastal_dofst_k_wfp_recovered_driver,/omc_data/whoi/OMC/CP02PMUO/R00015/instruments/CWFP*/*/C*.DA?,CP02PMUO-WFP01-02-DOFSTK000,recovered_wfp,Available,
+mi.dataset.driver.ctdpf_ckl.wfp.coastal_ctdpf_ckl_wfp_recovered_driver,/omc_data/whoi/OMC/CP02PMUO/R00015/instruments/CWFP*/*/C*.DA?,CP02PMUO-WFP01-03-CTDPFK000,recovered_wfp,Available,
+mi.dataset.driver.flort_kn.stc_imodem.flort_kn__stc_imodem_recovered_driver,/omc_data/whoi/OMC/CP02PMUO/R00015/instruments/CWFP*/*/E*.DA?,CP02PMUO-WFP01-04-FLORTK000,recovered_wfp,Available,
+mi.dataset.driver.PARAD_K.STC_IMODEM.parad_k_stc_imodem_recovered_driver,/omc_data/whoi/OMC/CP02PMUO/R00015/instruments/CWFP*/*/E*.DA?,CP02PMUO-WFP01-05-PARADK000,recovered_wfp,Available,
```
Comment (johnmlund) [inline CP02PMUO/CP02PMUO_R00015_ingest.csv:11]: nice!

File: CP03ISSM/CP03ISSM_R00011_ingest.csv (hunk 5 above)
Comment (johnmlund) [inline CP03ISSM/CP03ISSM_R00011_ingest.csv:43]: looks good

File: CP04OSSM/CP04OSSM_R00011_ingest.csv (hunk 6 above)
Comment (johnmlund) [inline CP04OSSM/CP04OSSM_R00011_ingest.csv:43]: looks good

**Review by johnmlund — APPROVED** [review]
all look ok to me.

**Pull request merged**